tuna/miopen/scripts/report.py

```diff
@@ -0,0 +1,254 @@
+
+    dfr = pd.merge(session_data,
+                   golden_data,
+                   on=['config', 'solver'],
+                   how='outer')
+    return dfr, session_data, golden_data
+
+
+def check_missing_configs(args, dbt, session_data, golden_data):
+  """Check for configs that are in the session tuning but not in the golden_v tuning"""
+  #get number of configs that are in golden_v but not in session
+
+  sess_configs = set(session_data['config'].unique().flatten().tolist())
+  golden_configs = set(golden_data['config'].unique().flatten().tolist())
+
+  sess_miss = sess_configs.difference(golden_configs)
+  gv_miss = golden_configs.difference(sess_configs)
+
+  print_driver_cmds(args, dbt, list(sess_miss), "Missing commands", 'session')
+  print_driver_cmds(args, dbt, list(gv_miss), "Missing commands", 'conv_golden')
+
+
+def print_driver_cmds(args, dbt, ids_list, text, table1):
+  """Print configs present in @table1 but missing from @table1"""
+  with DbSession() as session:
+    query = session.query(dbt.config_table.id, dbt.config_table.driver)\
+                   .filter(dbt.config_table.id.in_(ids_list))
+    drivers = query.all()
+    LOGGER.info("%s from %s:  %s", text, table1, len(drivers))
+    LOGGER.info("Driver cmds written to missing_%s_sess%s_gv%s.txt", table1,
+                args.session_id, args.golden_v)
+    missing_drivers = []
+    with open(f"missing_{table1}_sess{args.session_id}_gv{args.golden_v}.txt",
+              'w',
+              encoding='utf-8') as fout:
+      for entry in drivers:
+        if entry[1] is not None:
+          fout.write(f"{entry[1]}\n")
+        else:
+          missing_drivers.append(entry[0])
+    if missing_drivers:
+      LOGGER.warning(
+          'Configs with missing drivers in db: %s that could not be written to file',
+          len(missing_drivers))
+
+
+def summary_report(args, data_frame):
+  """Print tuning summary"""
+  #dfr cols = [id, config, solver, ktime_x, ktime_y]
+  dfr = data_frame.copy(deep=True)
+
+  #copy of only entires where ktime=-1
+  dfr_null = dfr.loc[dfr[['ktime_x', 'ktime_y']].eq(-1).any(axis=1)].copy()
+  dfr_null.to_csv('null.csv')
+  LOGGER.info("#configs with k_time=-1 in session_id= %s: %s", args.session_id,
+              dfr_null['ktime_x'].eq(-1).sum())
+  LOGGER.info("#configs with k_time=-1 in miopen_golden_v=%s : %s",
+              args.golden_v, dfr_null['ktime_y'].eq(-1).sum())
+
+  #remove entries where k_time=-1
+  dfr = dfr.loc[~dfr[['ktime_x', 'ktime_y']].eq(-1).any(axis=1)].copy()
+  dfr.to_csv('data.csv')
+
+  dfr['diff'] = dfr['ktime_x'] - dfr['ktime_y']
+  dfr.to_csv('data.csv')
+
+  #prct configs faster or slower
+  prct_positive = (dfr['diff'] > 0).sum() / dfr.shape[0] * 100
+  prct_equal = (dfr['diff'] == 0).sum() / dfr.shape[0] * 100
+  prct_negative = (dfr['diff'] < 0).sum() / dfr.shape[0] * 100
+  #pylint: disable=logging-format-truncated
+  LOGGER.info("configs with faster kernel_time: %f %%", round(prct_negative, 4))
```

Comment:
Either the text "configs with faster kernel_time" is misleading, or you are not doing what you intend. To find the configs that are faster in session_data, here's what I'd do:
0. Get rid of entries with -1 in both session_data and golden_data
1. Get the fastest solver (and it's time) for every config in session_data:
```
a = session_data.loc[session_data.groupby('config').ktime.idxmin()]
a = a.rename(columns={"solver": "s_sess", "ktime": "t_sess"})
```
2. Get the fastest solver (and it's time) for every config in  golden data:
```
b = golden_data.loc[golden_data.groupby('config').ktime.idxmin()]
b = b.rename(columns={"solver": "s_gold", "ktime": "t_gold"})
```
3. Join a and b on config
```
df = a.join(b, on="config")
```

And now, to find configs with faster kernel time, you can just do
```
(df["t_sess"] < df["t_gold"]).sum()
```

Reply:
I think this would also require changes to the rest of your analysis. I'll review the rest of the code after that

Reply:
you are right, this code doesnt do what it was intended to do anymore. I will move some of this analysis to the detailed_report() function where I can analyze config improvements based on fastest solver.